### ISSUE TITLE
Boot time variance too high?

### ISSUE THREAD

**Issue by dhrgit** [firecracker-microvm/firecracker#877]
The microvm boot time seems to be somewhat unstable, as shown by the values measured by `tests/integration_tests/performance/test_boottime.py`.

I ran the boot time test 100 times on an i3.metal and this is the data I gathered:
- mean: 105.3 ms
- median: 99.8 ms
- standard deviation: 8.6 ms
- min time: 98.7 ms
- max time: 120 ms

Also, there appear to be two large groups of values: the larger one gathered around 99ms, and a smaller one, gathered around 117ms. This might suggest a randomly occurring event, that adds about 18ms of boot time.

I think this requires further investigation.

**Comment by acatangiu** [comment]
Sure, it would be interesting to see **exactly** where the variance is coming from, but I don't think this is really worth investigating, 10ms outside standard deviation doesn't look worrisome.

**Comment by raduweiss** [comment]
That's a coefficient of variation of 8.2% ... I think it's actually quite good for a process starting on a non real-time system 😄.

I'm happy to even close this, until we start building aerospace software.

**Comment by dhrgit** [comment]
Yeah, I could've conveyed my point better than through those numbers. The more important issue was the grouping of the boot time values. I.e. two distinct groups (around 99ms and around 117ms. As I said, meaning that, occasionally, 18ms are lost somewhere we aren't aware of.
